### PR TITLE
Improve Makefile usage

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Add `help` option in `Makefile` to help developers. 
| Type?             |  improvement
| Category?         | CO 
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Run `make help` 
| Fixed ticket?     | 
| Related PRs       | 
| Sponsor company   | PrestaShop SA

Now, you can run `make help` to have a pretty help menu: 

Capture d’écran 2023-08-03 à 10 43 32

(about `.PHONY`, see https://stackoverflow.com/a/2145605)

### CODE REVIEW

**Review by boherm — DISMISSED** [review]
Nice improvement, thanks @nicosomb!

File: Makefile [hunk 1]
```diff
@@ -1,41 +1,48 @@
-install: composer assets
+help: ## Display this help menu
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-composer:
+install: composer assets  ## Install PrestaShop
```
Comment (M0rgan01) [inline Makefile]: The concept of installation is a bit vague here, we only install the dependencies, not prestashop itself.

File: Makefile [hunk 2]
```diff
@@ -1,41 +1,48 @@
-install: composer assets
+help: ## Display this help menu
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```
Comment (M0rgan01) [inline Makefile:2]: :heart_eyes:

File: Makefile (hunk 1 above)
Comment (nicosomb) [reply Makefile]: Oops it’s an error.

**Review by M0rgan01 — APPROVED** [review]

**Review by FabienPapet — APPROVED** [review]

**Comment by FabienPapet** [comment]
Well done !

**Pull request merged**